### ISSUE TITLE
A option to not use `use_frameworks!`

### ISSUE THREAD

**Issue by huanvu** [launchdarkly/react-native-client-sdk#13]
**Is this a support request?**
No

**Describe the bug**
`use_frameworks!` is often incompatible with other pods and even the react native repo admits so: https://github.com/react-native-community/releases/blob/master/CHANGELOG.md

In my case, its not compatible with react-native-firebase, most likely some others.

I've tried to not use it and instead add `:modular_headers => true` to  React and DarklyEventSource but get this instead:

```
▸ Check Dependencies
    no rule to process file '/Users/huanvu/Documents/hotlap-react/node_modules/launchdarkly-react-native-client-sdk/ios/LaunchdarklyReactNativeClient-Bridging-Header.h' of type sourcecode.c.h for architecture x86_64
error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening hotlap.xcworkspace
```

Not sure what to do here but if you have any instructions to use this sdk without `use_frameworks!` it would be of great help. Thank you!

**To reproduce**
Add `:modular_headers => true` to  React and DarklyEventSource in Podfile. 
Build.

**Expected behavior**
Project builds ok.

**Logs**
```
▸ Check Dependencies
    no rule to process file '/Users/huanvu/Documents/hotlap-react/node_modules/launchdarkly-react-native-client-sdk/ios/LaunchdarklyReactNativeClient-Bridging-Header.h' of type sourcecode.c.h for architecture x86_64
error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening hotlap.xcworkspace
```

**SDK version**
2.0.0

**Language version, developer tools**
React 16.8.6, React Native 0.59.9, XCode 10.2.1, Swift 4.2

**OS/platform**
OSX

**Additional context**
none

**Comment by torchhound** [comment]
Hi @huanvu, thanks for writing in! I'm not sure if we can support this directly, however the [LaunchDarkly iOS client can be used without frameworks](https://github.com/launchdarkly/ios-client-sdk#install-launchdarkly-without-frameworks). Please let us know if this works for you or contact support for more assistance.

**Comment by dkwl93** [comment]
Hey all, I have a similar issue. I'm trying to install LaunchDarkly pod via the line
`pod 'LaunchDarkly', :modular_headers => true, :git => 'https://github.com/launchdarkly/ios-client-sdk.git', :tag => '4.1.0'`

However, it is still giving me the error
```
The Swift pod `LaunchDarkly` depends upon `DarklyEventSource`, which do not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.
```

Can anyone shed some light on how to fix this? I am not using `use_frameworks` btw

Thank you!

CC: @huanvu

**Comment by dkwl93** [comment]
I've navigated around this issue by adding the following code to my podfile
```
  pre_install do |installer|
    # workaround for https://github.com/CocoaPods/CocoaPods/issues/3289
    Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_static_framework_transitive_dependencies) {}
  end
```

Hope this helps out anyone with the same issue

**Comment by phuochau** [comment]
I have same issue too, because we have same libraries that doesn't work with `use_frameworks` so looking forward to have a way to workaround so we can subscribe for the plan.

**Comment by ospfranco** [comment]
major issue when trying to use launch darkly, this should be a priority

**Comment by adrianso** [comment]
We would like to LaunchDarkly too, but unfortunately this is a show stopper for us.

**Comment by torchhound** [comment]
Hi all, thank you for letting us know about this requirement! We will be investigating removing `use_frameworks` from the iOS SDK.

**Comment by adrianso** [comment]
> Hi all, thank you for letting us know about this requirement! We will be investigating removing `use_frameworks` from the iOS SDK.

Do you have an ETA for that? I will need to decide whether I can afford to wait for that, or if I should move on to another product.

**Comment by torchhound** [comment]
@adrianso I'm sorry but we do not have a public ETA.

**Comment by alessioemireni** [comment]
Hi @torchhound, did you remove use_framework! after the 2.0.2 update?

**Comment by dkwl93** [comment]
Hello @torchhound do you know if upgrading our react-native version to 0.61 would fix this issue? 

> It also removes the React .xcodeproj, fixes use_frameworks! for CocoaPods support,

https://github.com/react-native-community/releases/blob/master/CHANGELOG.md

Thanks in advance for your hard work!

**Comment by torchhound** [comment]
@alessioemireni we are still using `use_frameworks`. @dkwl93 0.61 fixed the Pod issues with `use_frameworks` which is why we couldn't do a 0.60 release.

**Comment by ospfranco** [comment]
Just updated to 0.61 but this is still not working, any updates?

**Comment by theonetheycallneo** [comment]
We are having the same issue use_frameworks! conflicts with other pods in our file like react-native-maps + GoogleMaps.  Is there a work around if we are unable to use `use_frameworks!`?  Thanks in advance!

**Comment by lorenzoangelini** [comment]
Hi! @torchhound, when will you release 2.0.4? Will this problem be solved in 2.0.4?

**Comment by torchhound** [comment]
Hi all, as of version 2.1.0 `use_frameworks!` is no longer necessary. A big thanks to @sam1463, @hkellaway, and everyone who contributed ideas and pushed for this feature.
